### PR TITLE
Fix check of Literal in replace_pydantic_types

### PR DESCRIPTION
**Pull request by nina-j**
## Description

Make sure replace_pydantic_types returns early if the type found is a Literal - otherwise, the issubclass check fails later, because the `__args__` in Literals are vars and not types. 
Literal will fail to resolve later because Strawberry does not support them. 

Perhaps this should fail fast for unsupported Strawberry types that do not originate in Pydantic.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Nina for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

**Comment by nina-j** [comment]
`AssertionError: assert typing_extensions.Literal['field'] is typing_extensions.Literal['field']`

Ah yes. :(

File: strawberry/experimental/pydantic/object_type.py [hunk 1]
```diff
@@ -24,6 +25,9 @@
 
 
 def replace_pydantic_types(type_: Any):
+    origin = getattr(type_, "__origin__", None)
+    if origin is Literal:
+        return type_
```
Comment (patrick91) [inline strawberry/experimental/pydantic/object_type.py:30]: can we add a small comment on why we do this? ^^

**Review by patrick91 — COMMENTED** [review]
Looks great! I've only added a minor comment :)

RE: 

> Literal will fail to resolve later because Strawberry does not support them.

This is perfect, as we still allow devs to override the pydantic field type :)

**Comment by nina-j** [comment]
> ...
> This is perfect, as we still allow devs to override the pydantic field type :)

Oh, you do? Maybe I overlooked it in the documentation, but it hasn't been immediately clear to me that this was possible!

File: strawberry/experimental/pydantic/object_type.py (hunk 1 above)
Comment (nina-j) [reply strawberry/experimental/pydantic/object_type.py]: https://github.com/strawberry-graphql/strawberry/pull/1445/commits/2d43f7649491009ff359c976f500b76ffb12670d - hope this is sufficient!

**Comment by patrick91** [comment]
> Oh, you do? Maybe I overlooked it in the documentation, but it hasn't been immediately clear to me that this was possible!

Oh, we should document it better then!

you can do this:

```python
class Model(pydantic.BaseModel):
     field: Literal["field"]

 @strawberry.experimental.pydantic.type(Model)
 class Type:
     field: str
```

:)

**Review by patrick91 — APPROVED** [review]

**Pull request merged**